packages/components/namespace-store/reducer.tsx

```diff
@@ -0,0 +1,72 @@
+import { AWS_REGIONS } from "../../constants";
+
+export type ProviderDataState = {
+  secretName: string;
+  secretKey: string;
+  accessKey: string;
+  region: string;
+  target: string;
+  endpoint: string;
+};
+
+export const initialState: ProviderDataState = {
+  secretName: '',
+  secretKey: '',
+  accessKey: '',
+  region: AWS_REGIONS[0],
+  target: '',
+  endpoint: '',
+};
+
+export type StoreAction =
+  | { type: 'setSecretName'; value: string }
+  | { type: 'setSecretKey'; value: string }
+  | { type: 'setAccessKey'; value: string }
+  | { type: 'setRegion'; value: string }
+  | { type: 'setTarget'; value: string }
+  | { type: 'setEndpoint'; value: string };
+
+export type BackingStoreProviderDataState = ProviderDataState & {
+  numVolumes: number;
+  gcpJSON: string;
+  volumeSize: string;
+  storageClass: string;
+};
+
+export type BackingStoreAction =
+  | StoreAction
+  | { type: 'setGcpJSON'; value: string }
+  | { type: 'setVolumes'; value: number }
+  | { type: 'setVolumeSize'; value: string }
+  | { type: 'setStorageClass'; value: string };
+
+export const providerDataReducer = (
+  state: BackingStoreProviderDataState,
+  action: BackingStoreAction,
+) => {
+  const { value } = action;
+  switch (action.type) {
+    case 'setSecretName':
+      return Object.assign({}, state, { secretName: value });
+    case 'setSecretKey':
+      return Object.assign({}, state, { secretKey: value });
+    case 'setAccessKey':
+      return Object.assign({}, state, { accessKey: value });
+    case 'setRegion':
+      return Object.assign({}, state, { region: value });
+    case 'setGcpJSON':
```

Comment:
```suggestion
    case 'setGcpJSON':
```
why ??

Reply:
![image](https://user-images.githubusercontent.com/22158580/164649805-82546ecf-eb24-4984-a055-fb8445a4a44a.png)


initially, I copied from the console, that's why all these differences are there. later on, when bipul migrated to ODF he made those changes which is not reflected in my PR as I worked on the a week or two back

Reply:
`frontend/packages/ceph-storage-plugin/src/components/namespace-store/reducer.ts`
this is a reducer for namespacestore in console :)

Reply:
above screenshot is of reducer of backingstore